tools/git/git-2.31.yaml

```diff
@@ -0,0 +1,49 @@
+# metadata for git
+info:
+  id: tools/git
+  version: 2.31.0
+  priority: 20
+
+demands:
+  windows and x64:
+    unless:  # look for a system-installed version first
+      from: [$host.Path, $host.ProgramFiles/git/** ]            # paths to search
+      where:  git.exe                                       # one of these choices
+      run: '"$0" --version'                                           # run the choice with this cmdline
+      select: /git version (\d+\.\d+)\.\d*/ig                    # parse the output with this regex
+      is: "'2.31' === $1"                                          # this expression must be true.
+
+      # if all this succeeds, then the rest of this node will be used as the artifact
+      # substituting in $0 for the full path of the item that matches.
+      message: using discovered git $0
+      
+      settings: # if this is successful, these are the settings.
+        paths:
+          path: $0/.. # adds the found folder to the path
+        
+        tools:
+          git: $0
+     
+
+    # if we don't get a match, then we process the artifact as normal
+    # 
+    error: |
+      This requires git 2.31.* installed 

```

Comment:
Not for this PR, but will we eventually have a mechanism to localize messages like this?